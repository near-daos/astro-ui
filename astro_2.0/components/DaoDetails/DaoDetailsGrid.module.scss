@import 'styles/typography';
@import 'styles/mixins';

.root {
  @include card-shadow;
  height: 440px;
  max-width: 458px;
  min-width: 270px;
  transition: box-shadow 0.4s;
  width: 100%;

  &:hover {
    box-shadow: 3px 2px 24px var(--color-black-opacity-25);
  }
}

.content {
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: space-between;
  padding-bottom: 32px;
  width: 100%;
}

.general {
  align-items: center;
  display: flex;
}

.flag {
  flex: 0 0 69px;
  height: 78px;
  margin: 8px 10px;
  position: relative;
  width: 69px;

  @include tablet-portrait {
    margin: 20px 24px;
  }

  @include tablet-landscape {
    margin: 28px 24px;
  }

  .background {
    background-color: var(--color-white);
    clip-path: url('#flag');
    inset: 0;
    position: absolute;
    transform: scale(1.04, 1.045);
  }

  .cover {
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    clip-path: url('#flag');
    height: 100%;
    width: 100%;
  }

  .cover.scaled {
    transform: scale(1.45);
  }
}

.title {
  padding: 12px 16px 12px 0;
  width: 100%;

  @include tablet-portrait {
    padding-right: 24px;
  }
}

.name {
  @include font(28px, 28px, 700);
  @include ellipse-text;
  font-family: 'Roboto', 'sans-serif';

  @include astro-mobile {
    font-size: 20px;
  }
}

.nameTooltip {
  display: block;
  max-width: 200px;
  white-space: normal;
}

.address {
  @include font(16px, 24px, 500);
  align-items: center;
  color: var(--color-neutral-60);
  display: flex;
  margin-top: 8px;
  word-break: break-all;

  @include astro-mobile {
    font-size: 14px;
    > .addressId,
    > .copyAddress {
      @include ellipse-text;
      max-width: calc(70%);
    }
  }
}

.copyAddress {
  > button {
    margin-left: 6px;
  }
}

.description {
  @include font(16px, 24px, 400);
  height: 72px;
  overflow: hidden;
  padding: 0 24px;
}

.fundsAndMembers {
  display: flex;
  justify-content: space-around;
  padding: 0 24px;
}

.controls {
  @include centralize;
  justify-content: space-around;
  padding: 14px;

  @media (min-width: 480px) {
    justify-content: space-between;
    padding: 14px 28px;
  }

  @include tablet-landscape {
    justify-content: space-around;
    padding: 14px;
  }

  @include astro-desktop {
    justify-content: space-between;
    padding: 14px 28px;
  }

  @media (min-width: 1150px) {
    padding: 14px 20px;
  }

  @include large-desktop {
    padding: 14px 28px;
  }
}

.controlIcon {
  @include centralize;
  border-radius: 6px;
  transition: 0.2s linear;
  width: 42px;

  @media (min-width: 480px) {
    border-radius: 8px;
    width: auto;
  }

  @include tablet-landscape {
    border-radius: 6px;
    width: 42px;
  }

  @include astro-desktop {
    border-radius: 8px;
    width: auto;
  }

  > button {
    color: var(--color-neutral-70);
    width: 100%;

    @media (min-width: 480px) {
      @include font(10px, 15px, 500);
      flex-direction: row;
      padding: 0 14px;
    }

    @include tablet-landscape {
      @include font(8px, 12px, 400);
      flex-direction: column;
      padding: 0;
    }

    @include astro-desktop {
      @include font(10px, 15px, 500);
      flex-direction: row;
      padding: 0 14px;
    }
  }

  svg {
    width: 16px;

    @media (min-width: 480px) {
      margin: 0 6px 0 0;
    }

    @include tablet-landscape {
      margin: 0 0 4px;
    }

    @include astro-desktop {
      margin: 0 6px 0 0;
    }
  }
}

.controlIcon:hover {
  background-color: var(--color-neutral-30);
  color: var(--color-brand-black);

  > button {
    color: var(--color-brand-black);
  }
}

.proposals {
  @include centralize;
  flex-direction: column;
  padding: 0 16px;
  width: 100%;
}

.proposalsTitle {
  margin-bottom: 8px;

  &.active {
    font-size: 1.375em;
    font-weight: 300;
    line-height: 1.25em;
  }

  span {
    font-weight: 700;
  }
}

.proposalsSubtitle {
  color: var(--color-neutral-60);
  margin: 0;
  white-space: nowrap;
}
