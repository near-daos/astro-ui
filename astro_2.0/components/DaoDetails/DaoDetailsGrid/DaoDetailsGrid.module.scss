@import 'styles/typography';
@import 'styles/mixins';

.root {
  @include card-shadow;
  height: 440px;
  max-width: 458px;
  min-width: 270px;
  transition: box-shadow 0.4s;
  width: 100%;

  &:hover {
    box-shadow: 3px 2px 24px var(--color-black-opacity-25);
  }
}

.content {
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: space-between;
  padding-bottom: 32px;
  position: relative;
  width: 100%;
}

.general {
  align-items: center;
  display: flex;
}

.flagWrapper {
  border: 1px solid var(--color-black-opacity-1);
  border-radius: 50%;
  height: 68px;
  margin: 24px;
  min-height: 68px;
  min-width: 68px;
  overflow: hidden;
  position: relative;
  width: 68px;

  .flag {
    background-position: center;
    background-size: cover;
    height: 68px;
    width: 68px;
  }

  .flag.scaled {
    transform: scale(2.45);
  }

  .logo {
    @include centralize;
    background-color: var(--color-black);
    height: 100%;
    width: 100%;
  }
}

.title {
  padding: 12px 16px 12px 0;
  position: relative;
  width: 100%;

  @include tablet-portrait {
    padding-right: 24px;
  }
}

.explorerLink {
  position: absolute;
  right: 24px;
  top: 14px;
}

.name {
  @include font(28px, 28px, 700);
  @include ellipse-text;
  font-family: 'Roboto', 'sans-serif';
  line-height: 1.2;
  width: calc(100% - 36px);

  @include astro-mobile {
    font-size: 20px;
  }
}

.nameTooltip {
  display: block;
  max-width: 200px;
  white-space: normal;
}

.address {
  @include font(16px, 24px, 500);
  align-items: center;
  color: var(--color-neutral-60);
  display: flex;
  margin-top: 8px;
  word-break: break-all;

  @include tablet-portrait {
    display: block;
  }

  @include astro-mobile {
    font-size: 14px;
  }
}

.addressId {
  display: inline;
  margin-right: 6px;
}

.copyAddress {
  max-width: fit-content;
}

.description {
  @include font(16px, 24px, 400);
  height: 72px;
  overflow: hidden;
  padding: 0 24px;
}

.fundsAndMembers {
  display: flex;
  justify-content: space-around;
  padding: 0 24px;
}

.controls {
  @include centralize;
  justify-content: space-around;
  padding: 14px;

  @media (min-width: 480px) {
    justify-content: space-between;
    padding: 14px 28px;
  }

  @include tablet-landscape {
    justify-content: space-around;
    padding: 14px;
  }

  @include astro-desktop {
    justify-content: space-between;
    padding: 14px 28px;
  }

  @media (min-width: 1150px) {
    padding: 14px 20px;
  }

  @include large-desktop {
    padding: 14px 28px;
  }
}

.controlIcon {
  @include centralize;
  border-radius: 6px;
  transition: 0.2s linear;
  width: 42px;

  @media (min-width: 480px) {
    border-radius: 8px;
    width: auto;
  }

  @include tablet-landscape {
    border-radius: 6px;
    width: 42px;
  }

  @include astro-desktop {
    border-radius: 8px;
    width: auto;
  }

  > button {
    color: var(--color-neutral-70);
    width: 100%;

    @media (min-width: 480px) {
      @include font(10px, 15px, 500);
      flex-direction: row;
      padding: 0 14px;
    }

    @include tablet-landscape {
      @include font(8px, 12px, 400);
      flex-direction: column;
      padding: 0;
    }

    @include astro-desktop {
      @include font(10px, 15px, 500);
      flex-direction: row;
      padding: 0 14px;
    }
  }

  svg {
    width: 16px;

    @media (min-width: 480px) {
      margin: 0 6px 0 0;
    }

    @include tablet-landscape {
      margin: 0 0 4px;
    }

    @include astro-desktop {
      margin: 0 6px 0 0;
    }
  }
}

.controlIcon:hover {
  background-color: var(--color-neutral-30);
  color: var(--color-brand-black);

  > button {
    color: var(--color-brand-black);
  }
}

.proposals {
  @include centralize;
  flex-direction: column;
  padding: 0 16px;
  width: 100%;
}

.proposalsTitle {
  margin-bottom: 8px;

  &.active {
    font-size: 1.375em;
    font-weight: 300;
    line-height: 1.25em;
  }

  span {
    font-weight: 700;
  }
}

.proposalsSubtitle {
  color: var(--color-neutral-60);
  margin: 0;
  white-space: nowrap;
}
