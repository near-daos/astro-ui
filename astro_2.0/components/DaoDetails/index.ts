export * from './DaoDetailsMinimized';
export * from './DaoDetailsGrid';
