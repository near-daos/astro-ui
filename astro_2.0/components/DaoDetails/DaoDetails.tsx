import React, { FC } from 'react';
import cn from 'classnames';
import isEmpty from 'lodash/isEmpty';
import Link from 'next/link';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';

import { DAO } from 'types/dao';

import { ProposalTrackerCard } from 'astro_2.0/components/DaoDetails/components/ProposalTrackerCard';
import { DaoGeneralCard } from 'astro_2.0/components/DaoDetails/components/DaoGeneralCard';
import { ActionButton } from 'features/proposal/components/action-button';
import { isUserPermittedToCreateProposal } from 'astro_2.0/features/CreateProposal/createProposalHelpers';
import { getAccumulatedTokenValue } from 'features/treasury/helpers';
import { Token } from 'types/token';

import styles from './DaoDetails.module.scss';

export interface DaoDetailsProps {
  className?: string;
  dao: DAO;
  accountId?: string | null;
  onCreateProposalClick?: (dao: DAO) => void;
  activeProposals: number;
  totalProposals: number;
  restrictCreateProposals?: boolean;
  daoTokens: Record<string, Token>;
}

export const DaoDetails: FC<DaoDetailsProps> = ({
  dao,
  className,
  daoTokens,
  accountId,
  onCreateProposalClick,
  activeProposals,
  totalProposals,
  restrictCreateProposals = false,
}) => {
  const router = useRouter();
  const { t } = useTranslation();

  const isCreateOptionAvailable = isUserPermittedToCreateProposal(
    accountId,
    dao
  );

  const action =
    !isCreateOptionAvailable ||
    restrictCreateProposals ||
    isEmpty(accountId) ? null : (
      <>Create proposal</>
    );

  function getFundsInUsdFromNear() {
    const total = getAccumulatedTokenValue(daoTokens);

    return total.toFixed(2);
  }

  return (
    <div className={cn(styles.root, className)}>
      <section className={styles.general}>
        <DaoGeneralCard
          displayName={dao.displayName}
          id={dao.id}
          description={dao.description}
          links={dao.links}
          flag={dao.logo}
          cover={dao.flagCover}
          logo={dao.flagLogo}
          legal={dao.legal}
        />
      </section>

      <section className={styles.funds}>
        <Link href={`/dao/${dao.id}/treasury/tokens`}>
          <a>
            <div className={styles.label}>{t('daoFunds')}</div>
            <div className={styles.value}>
              <span className={styles.bold}>{getFundsInUsdFromNear()}</span> USD
            </div>
          </a>
        </Link>
      </section>

      <section className={styles.members}>
        <Link href={`/dao/${dao.id}/groups/all-members`}>
          <a>
            <div className={styles.label}>{t('membersGroups')}</div>
            <div className={styles.value}>
              <span className={styles.bold}>{dao.members}</span>/
              {dao.groups.length}
            </div>
          </a>
        </Link>
      </section>

      <section className={styles.controls}>
        <ActionButton
          tooltip={t('daoSettings')}
          tooltipPlacement="top"
          iconName="settings"
          onClick={() => {
            router.push(`/dao/${dao.id}/governance/settings`);
          }}
          className={styles.controlIcon}
        />
        <ActionButton
          tooltip={t('nfts')}
          tooltipPlacement="top"
          iconName="nfts"
          onClick={() => {
            router.push(`/dao/${dao.id}/treasury/nfts`);
          }}
          className={styles.controlIcon}
        />
        <ActionButton
          tooltip={t('bounties')}
          tooltipPlacement="top"
          iconName="proposalBounty"
          onClick={() => {
            router.push(`/dao/${dao.id}/tasks/bounties`);
          }}
          className={styles.controlIcon}
        />
        <ActionButton
          tooltip={t('polls')}
          tooltipPlacement="top"
          iconName="proposalPoll"
          onClick={() => {
            router.push(`/dao/${dao.id}/tasks/polls`);
          }}
          className={styles.controlIcon}
        />
      </section>

      <section className={styles.proposals}>
        <ProposalTrackerCard
          activeVotes={activeProposals}
          totalProposals={totalProposals}
          action={action}
          onClick={() => onCreateProposalClick?.(dao)}
        />
      </section>
    </div>
  );
};
