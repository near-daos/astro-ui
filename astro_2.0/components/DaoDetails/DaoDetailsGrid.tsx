import React, { FC } from 'react';
import Link from 'next/link';
import TextTruncate from 'react-text-truncate';
import { useRouter } from 'next/router';
import { useMeasure, useMedia } from 'react-use';
import { useTranslation } from 'next-i18next';
import cn from 'classnames';

import { GROUPS_PAGE_URL } from 'constants/routing';

import { DAO } from 'types/dao';

import { DaoInfoCard } from 'astro_2.0/components/DaoDetails/components/DaoInfoCard';
import { CopyButton } from 'astro_2.0/components/CopyButton';
import { ActionButton } from 'features/proposal/components/action-button';
import { formatCurrency } from 'utils/formatCurrency';
import * as Typography from 'components/Typography';
import { FlagRenderer } from 'astro_2.0/components/Flag';
import { Tooltip } from 'astro_2.0/components/Tooltip';

import { shortenString } from 'helpers/format';

import styles from './DaoDetailsGrid.module.scss';

export interface DaoDetailsGridProps {
  dao: DAO;
  activeProposals: number;
  totalProposals: number;
  nearPrice: number;
}

export const DaoDetailsGrid: FC<DaoDetailsGridProps> = ({
  dao,
  activeProposals,
  totalProposals,
  nearPrice,
}) => {
  const { t } = useTranslation();
  const router = useRouter();
  const {
    id,
    displayName,
    description,
    flagCover,
    logo: oldFlag,
    members,
    groups,
    funds,
  } = dao;
  const isMobile = useMedia('(max-width: 920px)');
  const fundsUSD = formatCurrency(parseFloat(funds) * nearPrice);

  const handleClick = (e: React.MouseEvent<HTMLButtonElement>, url: string) => {
    e.stopPropagation();
    e.preventDefault();
    router.push(url);
  };

  const [measureRef, { width }] = useMeasure<HTMLDivElement>();

  return (
    <div className={styles.root} ref={measureRef}>
      <Link href={`/dao/${id}`}>
        <a className={styles.content}>
          <div>
            <section className={styles.general}>
              <div className={styles.flag}>
                <FlagRenderer flag={flagCover} size="sm" fallBack={oldFlag} />
              </div>
              <div className={styles.title}>
                <Tooltip
                  placement="top"
                  overlay={
                    <span className={styles.nameTooltip}>
                      {displayName || id}
                    </span>
                  }
                >
                  <div
                    className={styles.name}
                    style={{
                      maxWidth: width - 142,
                    }}
                  >
                    {displayName || id}
                  </div>
                </Tooltip>
                <div className={styles.address}>
                  <div className={styles.addressId}>
                    {shortenString(id, isMobile ? 20 : 36)}
                  </div>
                  <CopyButton
                    text={id}
                    tooltipPlacement="auto"
                    className={styles.copyAddress}
                  />
                </div>
              </div>
            </section>

            <section className={styles.description}>
              <TextTruncate
                line={3}
                element="span"
                truncateText="â€¦"
                text={description ?? ''}
                textTruncateChild={null}
              />
            </section>
          </div>
          <div>
            <section className={styles.fundsAndMembers}>
              <DaoInfoCard
                infoType="funds"
                url={`/dao/${id}/treasury/tokens`}
                title={t('daoFunds')}
                daoFunds={fundsUSD}
                tooltip={t('daoFunds')}
              />
              <DaoInfoCard
                infoType="members"
                url={{
                  pathname: GROUPS_PAGE_URL,
                  query: {
                    dao: dao.id,
                    group: 'all',
                  },
                }}
                title={t('membersGroups')}
                members={members}
                groups={groups.length}
                tooltip={t('daoMembers')}
              />
            </section>

            <section className={styles.controls}>
              <ActionButton
                iconName="settings"
                onClick={e => handleClick(e, `/dao/${id}/governance/settings`)}
                className={styles.controlIcon}
              >
                {t('settings')}
              </ActionButton>
              <ActionButton
                iconName="nfts"
                onClick={e => handleClick(e, `/dao/${id}/treasury/nfts`)}
                className={styles.controlIcon}
              >
                {t('nfts')}
              </ActionButton>
              <ActionButton
                iconName="proposalBounty"
                onClick={e => handleClick(e, `/dao/${id}/tasks/bounties`)}
                className={styles.controlIcon}
              >
                {t('bounties')}
              </ActionButton>
              <ActionButton
                iconName="proposalPoll"
                onClick={e => handleClick(e, `/dao/${id}/tasks/polls`)}
                className={styles.controlIcon}
              >
                {t('polls')}
              </ActionButton>
            </section>

            <section className={styles.proposals}>
              <Typography.Subtitle
                className={cn(styles.proposalsTitle, styles.active)}
                size={2}
              >
                <span>{activeProposals}</span> {t('active')}{' '}
                {activeProposals === 1 ? t('proposal') : t('proposals')}
              </Typography.Subtitle>
              <Typography.Subtitle
                className={styles.proposalsSubtitle}
                size={6}
              >
                {`${totalProposals} ${t('proposalsInTotal')}`}
              </Typography.Subtitle>
            </section>
          </div>
        </a>
      </Link>
    </div>
  );
};
