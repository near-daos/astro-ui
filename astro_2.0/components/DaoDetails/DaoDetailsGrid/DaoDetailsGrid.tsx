import React, { FC } from 'react';
import Link from 'next/link';
import TextTruncate from 'react-text-truncate';
import { useRouter } from 'next/router';
import { useMeasure, useMedia } from 'react-use';
import { useTranslation } from 'next-i18next';
import cn from 'classnames';
import { UrlObject } from 'url';

import { ALL_BOUNTIES_PAGE_URL, GROUPS_PAGE_URL } from 'constants/routing';

import { DaoFeedItem } from 'types/dao';

import { CopyButton } from 'astro_2.0/components/CopyButton';
import { ActionButton } from 'astro_2.0/components/ActionButton';
import * as Typography from 'components/Typography';
import { Tooltip } from 'astro_2.0/components/Tooltip';
import { Icon } from 'components/Icon';

import { shortenString } from 'utils/format';
import { formatCurrency } from 'utils/formatCurrency';

import { ExplorerLink } from 'components/ExplorerLink';
import { DaoInfoCard } from './components/DaoInfoCard';
import { DaoDetailsSkeleton } from './components/DaoDetailsSkeleton';

import styles from './DaoDetailsGrid.module.scss';

export interface DaoDetailsGridProps {
  dao: DaoFeedItem;
  activeProposals: number;
  totalProposals: number;
  loading?: boolean;
}

export const DaoDetailsGrid: FC<DaoDetailsGridProps> = ({
  dao,
  activeProposals,
  totalProposals,
  loading,
}) => {
  const { t } = useTranslation();
  const router = useRouter();
  const {
    id,
    displayName,
    description,
    flagLogo,
    logo: oldFlag,
    numberOfMembers,
    numberOfGroups,
    totalDaoFunds,
  } = dao;
  const isMobile = useMedia('(max-width: 920px)');

  const handleClick = (
    e: React.MouseEvent<HTMLButtonElement>,
    url: string | UrlObject
  ) => {
    e.stopPropagation();
    e.preventDefault();
    router.push(url);
  };

  const [measureRef, { width }] = useMeasure<HTMLDivElement>();

  return (
    <div className={styles.root} ref={measureRef}>
      {loading ? (
        <DaoDetailsSkeleton />
      ) : (
        <Link href={`/dao/${id}`}>
          <a className={styles.content}>
            <div>
              <section className={styles.general}>
                <div className={styles.flagWrapper}>
                  {flagLogo ? (
                    <div
                      className={cn(styles.flag, {
                        [styles.scaled]: !flagLogo && !!oldFlag,
                      })}
                      style={{
                        backgroundImage: `url(${flagLogo || oldFlag})`,
                      }}
                    />
                  ) : (
                    <div className={styles.logo}>
                      <Icon
                        name="defaultDaoLogo"
                        className={styles.defaultLogo}
                        width={24}
                      />
                    </div>
                  )}
                </div>
                <div className={styles.title}>
                  <ExplorerLink
                    linkData={dao.id}
                    linkType="member"
                    className={styles.explorerLink}
                  />
                  <Tooltip
                    placement="top"
                    overlay={
                      <span className={styles.nameTooltip}>
                        {displayName || id}
                      </span>
                    }
                  >
                    <div
                      className={styles.name}
                      style={{
                        maxWidth: width - 142,
                      }}
                    >
                      {displayName || id}
                    </div>
                  </Tooltip>
                  <div className={styles.address}>
                    <CopyButton
                      text={id}
                      tooltipPlacement="auto"
                      className={styles.copyAddress}
                    >
                      <div className={styles.addressId}>
                        {shortenString(id, isMobile ? 16 : 26)}
                      </div>
                    </CopyButton>
                  </div>
                </div>
              </section>

              <section className={styles.description}>
                <TextTruncate
                  line={3}
                  element="span"
                  truncateText="â€¦"
                  text={description ?? ''}
                  textTruncateChild={null}
                />
              </section>
            </div>
            <div>
              <section className={styles.fundsAndMembers}>
                <DaoInfoCard
                  infoType="funds"
                  url={`/dao/${id}/treasury/tokens`}
                  title={t('daoFunds')}
                  daoFunds={formatCurrency(Number(totalDaoFunds ?? 0))}
                  tooltip={t('daoFunds')}
                />
                <DaoInfoCard
                  infoType="members"
                  url={{
                    pathname: GROUPS_PAGE_URL,
                    query: {
                      dao: dao.id,
                      group: 'all',
                    },
                  }}
                  title={t('membersGroups')}
                  members={numberOfMembers}
                  groups={numberOfGroups}
                  tooltip={t('daoMembers')}
                />
              </section>

              <section className={styles.controls}>
                <ActionButton
                  iconName="settings"
                  onClick={e =>
                    handleClick(e, `/dao/${id}/governance/settings`)
                  }
                  className={styles.controlIcon}
                >
                  {t('settings')}
                </ActionButton>
                <ActionButton
                  iconName="nfts"
                  onClick={e => handleClick(e, `/dao/${id}/treasury/nfts`)}
                  className={styles.controlIcon}
                >
                  {t('nfts')}
                </ActionButton>
                <ActionButton
                  iconName="proposalBounty"
                  onClick={e =>
                    handleClick(e, {
                      pathname: ALL_BOUNTIES_PAGE_URL,
                      query: {
                        dao: id,
                      },
                    })
                  }
                  className={styles.controlIcon}
                >
                  {t('bounties')}
                </ActionButton>
                <ActionButton
                  iconName="proposalPoll"
                  onClick={e => handleClick(e, `/dao/${id}/tasks/polls`)}
                  className={styles.controlIcon}
                >
                  {t('polls')}
                </ActionButton>
              </section>

              <section className={styles.proposals}>
                <Typography.Subtitle
                  className={cn(styles.proposalsTitle, styles.active)}
                  size={2}
                >
                  <span>{activeProposals}</span>{' '}
                  {activeProposals === 1
                    ? t('components.daoDetails.proposalTrackerCard.active')
                    : t(
                        'components.daoDetails.proposalTrackerCard.activePlural'
                      )}{' '}
                  {activeProposals === 1
                    ? t('components.daoDetails.proposalTrackerCard.proposal')
                    : t('components.daoDetails.proposalTrackerCard.proposals')}
                </Typography.Subtitle>
                <Typography.Subtitle
                  className={styles.proposalsSubtitle}
                  size={6}
                >
                  {`${totalProposals} ${t(
                    'components.daoDetails.proposalTrackerCard.proposalsInTotal'
                  )}`}
                </Typography.Subtitle>
              </section>
            </div>
          </a>
        </Link>
      )}
    </div>
  );
};
