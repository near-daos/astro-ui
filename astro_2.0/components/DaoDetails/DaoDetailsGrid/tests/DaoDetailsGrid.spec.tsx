/* eslint-disable @typescript-eslint/ban-ts-comment */

import { render } from 'jest/testUtils';
import { useRouter } from 'next/router';
import { fireEvent } from '@testing-library/dom';

import { DaoFeedItem } from 'types/dao';

import { DaoDetailsGrid } from 'astro_2.0/components/DaoDetails/DaoDetailsGrid';

import { daoMock } from './mock';

jest.mock('react-use', () => {
  return {
    ...jest.requireActual('react-use'),
    useMedia: jest.fn(),
    useMeasure: jest.fn(() => [() => 0, { width: 10 }]),
  };
});

jest.mock('react-text-truncate', () => {
  return () => <div />;
});

jest.mock(
  'astro_2.0/components/DaoDetails/DaoDetailsGrid/components/DaoDetailsSkeleton',
  () => ({
    DaoDetailsSkeleton: () => <div>Skeleton</div>,
  })
);

jest.mock('next/router', () => {
  return {
    useRouter: jest.fn(),
  };
});

describe('dao details grid', () => {
  it('Should render component', () => {
    const { container } = render(
      <DaoDetailsGrid dao={daoMock} activeProposals={12} totalProposals={12} />
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render skeleton if data is loading', () => {
    const { container } = render(
      <DaoDetailsGrid
        dao={daoMock}
        activeProposals={12}
        totalProposals={12}
        loading
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('Should use dao id when displayName not presented', () => {
    const id = 'myTestId';

    const dao = {
      ...daoMock,
      displayName: undefined,
      description: undefined,
      id,
    } as unknown as DaoFeedItem;

    const component = render(
      <DaoDetailsGrid dao={dao} activeProposals={12} totalProposals={12} />
    );

    expect(component.getAllByText(id)).toHaveLength(2);
  });

  it.each`
    button        | url
    ${'settings'} | ${`/dao/${daoMock.id}/governance/settings`}
    ${'nfts'}     | ${`/dao/${daoMock.id}/treasury/nfts`}
    ${'bounties'} | ${`/dao/${daoMock.id}/tasks/bounties/list`}
    ${'polls'}    | ${`/dao/${daoMock.id}/tasks/polls`}
  `(
    'Should navigate to proper url when user click "$button" action button',
    ({ button }) => {
      const router = {
        push: jest.fn(),
      };

      // @ts-ignore
      useRouter.mockImplementation(() => router);

      const component = render(
        <DaoDetailsGrid
          dao={daoMock}
          activeProposals={12}
          totalProposals={12}
        />
      );

      const actionButton = component.getByText(button);

      fireEvent.click(actionButton);

      expect(router.push).toBeCalled();
    }
  );

  it('Should use singular titles for proposals when needed', () => {
    const { getAllByText } = render(
      <DaoDetailsGrid dao={daoMock} activeProposals={1} totalProposals={1} />
    );

    expect(
      getAllByText('components.daoDetails.proposalTrackerCard.active', {
        exact: false,
      })
    ).toHaveLength(1);

    expect(
      getAllByText(/components.daoDetails.proposalTrackerCard.proposal$/, {
        exact: false,
      })
    ).toHaveLength(1);
  });
});

/* eslint-enable @typescript-eslint/ban-ts-comment */
