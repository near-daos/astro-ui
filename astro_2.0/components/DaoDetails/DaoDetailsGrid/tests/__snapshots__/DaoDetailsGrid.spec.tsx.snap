// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dao details grid Should render component 1`] = `
<div>
  <div
    class="root"
  >
    <a
      class="content"
      href="/dao/default-flag-test.sputnikv2.testnet"
    >
      <div>
        <section
          class="general"
        >
          <div
            class="flagWrapper"
          >
            <div
              class="logo"
            >
              <svg
                class="defaultLogo"
                width="24"
              >
                <use
                  href="/_next/undefined"
                />
              </svg>
            </div>
          </div>
          <div
            class="title"
          >
            <div
              class="explorerLink"
            >
              <div
                class="root"
                role="button"
                tabindex="0"
              >
                <span
                  class="iconWrapper"
                >
                  <svg
                    class="icon"
                  >
                    <use
                      href="/_next/undefined"
                    />
                  </svg>
                </span>
              </div>
            </div>
            <div
              class=""
            >
              <div
                class="name"
                style="max-width: -132px;"
              >
                Default Flag Test
              </div>
            </div>
            <div
              class="address"
            >
              <div
                class="root copyAddress"
                role="button"
                tabindex="0"
              >
                <div
                  class="addressId"
                >
                  default-flag-test.sput...stnet
                </div>
                <button
                  class="iconButton small btn"
                  type="button"
                >
                  <svg>
                    <use
                      href="/_next/undefined"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </section>
        <section
          class="description"
        >
          <div />
        </section>
      </div>
      <div>
        <section
          class="fundsAndMembers"
        >
          <div
            class="root"
          >
            <button
              class="btn transparent sizeBlock"
              type="button"
            >
              <div
                class="label"
              >
                daoFunds
              </div>
              <div
                class="value"
              >
                <span
                  class="bold"
                >
                  97.30
                </span>
                 USD
              </div>
            </button>
          </div>
          <div
            class="root members"
          >
            <button
              class="btn transparent sizeBlock"
              type="button"
            >
              <div
                class="label"
              >
                membersGroups
              </div>
              <div
                class="value"
              >
                <span
                  class="bold"
                />
                /undefined
              </div>
            </button>
          </div>
        </section>
        <section
          class="controls"
        >
          <div
            class="controlIcon"
          >
            <button
              class="iconTextButton"
              type="button"
            >
              <svg
                class="icon"
              >
                <use
                  href="/_next/undefined"
                />
              </svg>
              settings
            </button>
          </div>
          <div
            class="controlIcon"
          >
            <button
              class="iconTextButton"
              type="button"
            >
              <svg
                class="icon"
              >
                <use
                  href="/_next/undefined"
                />
              </svg>
              nfts
            </button>
          </div>
          <div
            class="controlIcon"
          >
            <button
              class="iconTextButton"
              type="button"
            >
              <svg
                class="icon"
              >
                <use
                  href="/_next/undefined"
                />
              </svg>
              bounties
            </button>
          </div>
          <div
            class="controlIcon"
          >
            <button
              class="iconTextButton"
              type="button"
            >
              <svg
                class="icon"
              >
                <use
                  href="/_next/undefined"
                />
              </svg>
              polls
            </button>
          </div>
        </section>
        <section
          class="proposals"
        >
          <p
            class="subtitle2 proposalsTitle active"
          >
            <span>
              12
            </span>
             
            components.daoDetails.proposalTrackerCard.activePlural
             
            components.daoDetails.proposalTrackerCard.proposals
          </p>
          <p
            class="subtitle6 proposalsSubtitle"
          >
            12 components.daoDetails.proposalTrackerCard.proposalsInTotal
          </p>
        </section>
      </div>
    </a>
  </div>
</div>
`;

exports[`dao details grid Should render skeleton if data is loading 1`] = `
<div>
  <div
    class="root"
  >
    <div>
      Skeleton
    </div>
  </div>
</div>
`;
